src/components/molecules/Main/Main.tsx

```diff
@@ -0,0 +1,36 @@
+import React, {useEffect} from 'react';
+import {CustomButton} from '../../atoms';
+import {View} from 'react-native';
+import {useNavigation} from '@react-navigation/native';
+import {ROUTES, RootStackParamList} from '../../../static/types/routeTypes';
+import {StackNavigationProp} from '@react-navigation/stack';
+import {globalStyles} from '../../../utils/globalStyles';
+
+type TMainProps = {
+  setIndex: React.Dispatch<React.SetStateAction<number>>;
+};
+export const Main = ({setIndex}: TMainProps) => {
+  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();
+
+  useEffect(() => {
+    if (navigation.isFocused()) {
```

Comment:
Jest coś takiego jak hook useIsFocused (jakoś tak), który zwraca ci value czy ekran jest focused czy nie

 A jeżeli zależy ci na effekcie samym, to w RN masz coś takiego jak useFocusEffect, działa podobnie jak useEffect, ale reaguje na focusScreena i opuszczenie go, a nie na mount i unmount
 
 
Hint, do środka useFocusEffect musisz useCallback wrzucić aby mieć tablicę depsów itp.

Reply:
done